File: client-scalaz/src/main/scala/org/openapitools/client/api/ReportResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ReportResponse._

case class ReportResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
columns: Option[List[ReportTypeResponse]],
rows: Option[List[Map[String, Any]]],
summations: Option[Map[String, BigDecimal]],
count: Option[Long],
queryName: Option[String],
hasMore: Option[Boolean],
returning: Option[String])

object ReportResponse {
  import DateTimeCodecs._

  implicit val ReportResponseCodecJson: CodecJson[ReportResponse] = CodecJson.derive[ReportResponse]
  implicit val ReportResponseDecoder: EntityDecoder[ReportResponse] = jsonOf[ReportResponse]
  implicit val ReportResponseEncoder: EntityEncoder[ReportResponse] = jsonEncoderOf[ReportResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/report_response.clj
(ns sirqul-io-t-platform.specs.report-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.report-type-response :refer :all]
            )
  (:import (java.io File)))


(def report-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :columns) (s/coll-of report-type-response-spec)
   (ds/opt :rows) (s/coll-of (s/map-of string? any?))
   (ds/opt :summations) (s/map-of string? float?)
   (ds/opt :count) int?
   (ds/opt :queryName) string?
   (ds/opt :hasMore) boolean?
   (ds/opt :returning) string?
   })

(def report-response-spec
  (ds/spec
    {:name ::report-response
     :spec report-response-data}))
